### ISSUE TITLE
Syntax Error in function after build with mongo library

### ISSUE THREAD

**Issue by rendall** [netlify/netlify-lambda#135]
TL;DR: Is there a babel setting that might compile source code to a pre- function-generator version? I'm not sure why, but the hosted function emits a Syntax Error related to a generator function when I include the Nodejs MongoDB driver.

----

When I include the native [MongoDB Node.js driver](http://mongodb.github.io/node-mongodb-native/), the function compiles and serves fine locally, but blows up after upload, as seen here: 
https://focused-elion-17290b.netlify.com/.netlify/functions/connect

The error message is:
`"Unexpected token *"` and the function log is:

```
Syntax error in module 'connect': SyntaxError
async function* asyncIterator() {
              ^
```

It appears that the Netlify (or AWS?) Node environment cannot parse generator functions, although the Node version is according to the build log is (as I specified):
`Now using node v11.9.0 (npm v6.5.0)`

Is fixing this accessible to me? Perhaps a babel setting for `netlify-lambda` that compiles to a pre-generator version of javascript?  Any other ideas?

The 'connect' endpoint above is stripped-down and taken directly from the [MongoDB website](https://www.mongodb.com/blog/post/serverless-development-with-nodejs-aws-lambda-mongodb-atlas), and as I said, it works as expected when served locally via `netlify-lambda serve`. If it might be helpful, the 'connect' endpoint [source code is here](https://github.com/rendall/api-cat/blob/56736e31206cefd74d9752bf86490b971bd1bd15/ts/functions-src/connect.ts).

**Comment by sw-yx** [comment]
i mean theres a babel transform for everything :) look around.. i found this? https://babeljs.io/docs/en/babel-plugin-proposal-async-generator-functions

and just supply your own .babelrc: https://github.com/netlify/netlify-lambda#webpack-configuration

**Comment by AllenAnthes** [comment]
Ran into the same issue today when upgrading mongoose from 5.4.21 to 5.5.0.

Looks like the root cause for me was mongoose bumping their mongodb dependency from 3.1.13 to 3.2.2.

Either way, for anyone else running into the issue pinning mongoose back to 5.4 as a quick/temporary fix worked for me.

**Comment by rendall** [comment]
Hey that's really useful information. Thanks!

**Comment by rendall** [comment]
@AllenAnthes was correct: the 3.2.2 mongodb library is the problem. After downgrading the version back to 3.1.13 the function works as expected. I suggest this for anyone who runs into this issue.

Thanks to everyone who commented.

**Comment by sw-yx** [comment]
huh. cool. i hope the mongoose repo has an issue reporting the same.
